pkg/kubelet/volumes.go

```diff
@@ -366,45 +382,45 @@ func (kl *Kubelet) newVolumeMounterFromPlugins(spec *volume.Spec, pod *api.Pod,
 // newVolumeAttacherFromPlugins attempts to find a plugin from a volume spec
 // and then create an Attacher.
 // Returns:
-//  - an attacher if one exists
+//  - an attacher if one exists, nil otherwise
+//  - the AttachableVolumePlugin if attacher exists, nil otherewise
 //  - an error if no plugin was found for the volume
-//    or the attacher failed to instantiate
-//  - nil if there is no appropriate attacher for this volume
-func (kl *Kubelet) newVolumeAttacherFromPlugins(spec *volume.Spec, pod *api.Pod) (volume.Attacher, error) {
+//    or the attacher failed to instantiate, nil otherwise
+func (kl *Kubelet) newVolumeAttacherFromPlugins(spec *volume.Spec, pod *api.Pod) (volume.Attacher, volume.AttachableVolumePlugin, error) {
 	plugin, err := kl.volumePluginMgr.FindAttachablePluginBySpec(spec)
 	if err != nil {
-		return nil, fmt.Errorf("can't use volume plugins for %s: %v", spec.Name(), err)
+		return nil, nil, fmt.Errorf("can't use volume plugins for %s: %v", spec.Name(), err)
 	}
 	if plugin == nil {
 		// Not found but not an error.
-		return nil, nil
+		return nil, nil, nil
 	}
 
 	attacher, err := plugin.NewAttacher()
 	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate volume attacher for %s: %v", spec.Name(), err)
+		return nil, nil, fmt.Errorf("failed to instantiate volume attacher for %s: %v", spec.Name(), err)
 	}
 	glog.V(3).Infof("Used volume plugin %q to attach %s/%s", plugin.Name(), spec.Name())
```

Comment:
I hate how these messages say 'used' - it implies that the work is already done.  change to `using`?